src/Role/Repository/RoleHierarchyRepository.php

```diff
@@ -90,6 +108,71 @@ public function hasChildRoleId(int $parentRoleId, int $findingChildId): bool
         return false;
     }
 
+     /**
+     * @param int $parentRoleId
+     * @param int $findingParentId
+     *
+     * @return bool
+     * @throws QueryException
+     */
+    public function hasParentRoleId(int $parentRoleId, int $findingParentId): bool
+    {
+        $parentIds = $this->getParentIds([$parentRoleId]);
+
+        if(count($parentIds) > 0) {
+            if(in_array($findingParentId, $parentIds, true)) {
+                return true;
+            }
+
+            foreach($parentIds as $parentId) {
+                if($this->hasParentRoleId($parentId, $findingParentId)) {
+                    return true;
+                }
+            }
+        }
+
+        return false;
+    }
+
+    public function areBrothers(int $parentRoleId, int $childRoleId) : bool {
```

Comment:
changes are looking good please add a docblock over the function that describes what it does